lib/ASTGen/Sources/ASTGen/Types.swift

```diff
@@ -278,7 +278,7 @@ extension ASTGenVisitor {
       ),
       asyncLoc: (node.effectSpecifiers?.asyncSpecifier).bridgedSourceLoc(in: self),
       throwsLoc: (node.effectSpecifiers?.throwsSpecifier).bridgedSourceLoc(in: self),
-      thrownType: self.generate(optional: node.effectSpecifiers?.thrownError?.type).asNullable,
+      thrownType: self.map(node.effectSpecifiers?.thrownError?.type, generate(type:)),
```

Comment:
I’m not a fan of this. I think the old version read a lot easier and IMO it read so much easier that it’s worth defining the optional overloads.

@AnthonyLatsis and I had a whole conversation about this in https://github.com/apple/swift/pull/67111#discussion_r1258123567

Reply:
I don't feel too strongly about this, though I think I do mildly prefer the old version

Reply:
OK, let's restore the specialized versions.